src/ui/public/visualize/loader/pipeline_helpers.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import { fromExpression } from '@kbn/interpreter/common';
+import { interpretAst } from '@kbn/interpreter/public';
+
+const getSchemas = (vis) => {
+  let cnt = 0;
+  const schemas = {
+    metric: [],
+  };
+  const responseAggs = vis.aggs.getResponseAggs();
+  const isHierarchical = vis.isHierarchical();
+  const metrics = responseAggs.filter(agg => agg.type.type === 'metrics');
+  responseAggs.forEach((agg) => {
+    if (!agg.enabled) return;
+    let schemaName = agg.schema ? agg.schema.name || agg.schema : null;
+    if (typeof schemaName === 'object') schemaName = null;
+    if (!schemaName) return;
```

Comment:
no, thats not true, agg.schema might be a string. it might also be an object, in that case it should have a .name property on it .... if it doesnt (and its an object), we need to trat it as null.